File: Selection Sort/selectionSort.cpp
/*
    @author: Guilherme Vieira Silva Gonçalves
    @course: Ciência da Computação, 5th semester
*/

#include <iostream>

void selectionSort(int arr[], int n){

    int i, j, min;
    for (i = 0; i < n - 1; i++) {
        min = i;
        for (j = i+1; j < n; j++) if (arr[j] < arr[min]) min = j;
        int aux = arr[min];
        arr[min] = arr[i];
        arr[i] = aux;

    }

}

int main() {
    int arr[] = {10, 7, 3, 8, 4, 2, 5};
    int n = sizeof(arr) / sizeof(arr[0]);
    selectionSort(arr, n);
    for (int i = 0; i < n; i++) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;


}

File: Selection Sort/selectionSort.py
'''
    @author: Guilherme Vieira Silva Gonçalves
    @course: Ciência da Computação, 5th semester
'''

def selectionSort(arr):
    n = len(arr)
    
    for i in range(0, n - 1):
        min = i
        for j in range(i+1, n):
            if arr[j] < arr[min]:
                min = j
        arr[min], arr[i] = arr[i], arr[min]
    return arr


def main():
    arr = [3,2,1,6]
    print(selectionSort(arr))

if __name__ == "__main__":
    main()
